src/scale/format.rs

```diff
@@ -1,214 +0,0 @@
-	U8,
-	/// Equivalent to [`u16`].
-	U16,
-	/// Equivalent to [`u32`].
-	U32,
-	/// Equivalent to [`u64`].
-	U64,
-}
-
-/// An error that can occur when we try to encode or decode to a SCALE bit sequence type.
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub enum FromMetadataError {
-	/// The registry did not contain the bit order type listed.
-	OrderFormatNotFound(u32),
-	/// The registry did not contain the bit store type listed.
-	StoreFormatNotFound(u32),
-	/// The bit order type did not have a valid identifier/name.
-	NoBitOrderIdent,
-	/// The bit store type that we found was not what we expected (a primitive u8/u16/u32/u64).
-	StoreFormatNotSupported(String),
-	/// The bit order type name that we found was not what we expected ("Lsb0" or "Msb0").
-	OrderFormatNotSupported(String),
-}
-
-impl core::fmt::Display for FromMetadataError {
-	fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-		match self {
-			FromMetadataError::OrderFormatNotFound(n) => {
-				write!(f, "Bit order type {n} not found in registry")
-			}
-			FromMetadataError::StoreFormatNotFound(n) => {
-				write!(f, "Bit store type {n} not found in registry")
-			}
-			FromMetadataError::NoBitOrderIdent => {
-				write!(f, "Bit order cannot be identified")
-			}
-			FromMetadataError::StoreFormatNotSupported(s) => {
-				write!(f, "Bit store type '{s}' is not supported")
-			}
-			FromMetadataError::OrderFormatNotSupported(s) => {
-				write!(f, "Bit order type '{s}' is not supported")
-			}
-		}
-	}
-}
-
-#[cfg(feature = "std")]
-impl std::error::Error for FromMetadataError {}
-
-#[cfg(feature = "scale-info")]
-#[cfg(test)]
-mod test {
-	use super::*;
-
-	fn make_type<T: scale_info::TypeInfo + 'static>() -> (u32, scale_info::PortableRegistry) {
-		let m = scale_info::MetaType::new::<T>();
-		let mut types = scale_info::Registry::new();
-		let id = types.register_type(&m);
-		let portable_registry: scale_info::PortableRegistry = types.into();
-
-		(id.id, portable_registry)
-	}
-
-	fn assert_format<T: scale_info::TypeInfo + 'static>(store: StoreFormat, order: OrderFormat) {
-		// Encode to metadata:
-		let (id, types) = make_type::<T>();
-
-		// Pull out said type info:
-		let ty = match &types.resolve(id).unwrap().type_def {
-			scale_info::TypeDef::BitSequence(b) => b,
-			_ => panic!("expected type to look like a bit sequence"),
-		};
-
-		// We should be able to obtain a valid Format from it:
-		let actual_format =
-			crate::Format::from_metadata(ty, &types).expect("can obtain BitSeq Format from type");
-
-		// The format should match the one we expect:
-		assert_eq!(Format::new(store, order), actual_format);
-	}
-
-	#[test]
-	fn format_extracted_properly() {
-		use bitvec::{
-			order::{Lsb0, Msb0},
-			vec::BitVec,
-		};
-
-		assert_format::<crate::Bits>(StoreFormat::U8, OrderFormat::Lsb0);
-
-		assert_format::<BitVec<u8, Lsb0>>(StoreFormat::U8, OrderFormat::Lsb0);
-		assert_format::<BitVec<u16, Lsb0>>(StoreFormat::U16, OrderFormat::Lsb0);
```

Comment:
Nice! These live now under [scale-type-resolver](https://github.com/paritytech/scale-type-resolver/blob/c40e3f213a78179e2a65e005d30b3d34dbd26a16/src/portable_registry.rs#L402-L440)

Reply:
Yup; first I moved them to scale-type-resolver and then I deleted them in favour of more generic resolve_type tests that cover this stuff anyways :)